crates/factor-variables/src/spin_cli/azure_key_vault.rs

```diff
@@ -0,0 +1,157 @@
+use std::sync::Arc;
+
+use anyhow::Context as _;
+use azure_core::{auth::TokenCredential, Url};
+use azure_security_keyvault::SecretClient;
+use serde::Deserialize;
+use spin_expressions::{Key, Provider};
+use spin_factors::anyhow;
+use spin_world::async_trait;
+use tracing::{instrument, Level};
+
+/// Azure KeyVault runtime config literal options for authentication
+///
+/// Some of these fields are optional. Whether they are set determines whether environmental variables
+/// will be used to resolve the information instead.
+#[derive(Clone, Debug, Deserialize)]
+#[serde(deny_unknown_fields)]
+pub struct AzureKeyVaultVariablesConfig {
+    pub vault_url: String,
+    pub client_id: Option<String>,
+    pub client_secret: Option<String>,
+    pub tenant_id: Option<String>,
+    pub authority_host: Option<AzureAuthorityHost>,
+}
+
+#[derive(Debug, Copy, Clone, Deserialize, Default)]
+pub enum AzureAuthorityHost {
```

Comment:
This is probably out of scope for _this_ PR but this seems like an odd enum to maintain here...